backend/tests/integration/common_utils/managers/user.py

```diff
@@ -101,41 +113,118 @@ def verify_role(user_to_verify: DATestUser, target_role: UserRole) -> bool:
     def set_role(
         user_to_set: DATestUser,
         target_role: UserRole,
-        user_to_perform_action: DATestUser | None = None,
-    ) -> None:
-        if user_to_perform_action is None:
-            user_to_perform_action = user_to_set
+        user_performing_action: DATestUser | None = None,
+    ) -> DATestUser:
+        if user_performing_action is None:
+            user_performing_action = user_to_set
         response = requests.patch(
             url=f"{API_SERVER_URL}/manage/set-user-role",
             json={"user_email": user_to_set.email, "new_role": target_role.value},
-            headers=user_to_perform_action.headers,
+            headers=user_performing_action.headers,
+        )
+        response.raise_for_status()
+        user_to_set.role = target_role
+        return user_to_set
+
+    @staticmethod
+    def verify_status(
+        user_to_verify: DATestUser, target_status: UserStatus | None = None
+    ) -> bool:
+        if target_status is None:
+            target_status = user_to_verify.status
+        response = requests.get(
+            url=f"{API_SERVER_URL}/me",
+            headers=user_to_verify.headers,
+        )
+        response.raise_for_status()
+
+        is_active = response.json().get("is_active", None)
+        if is_active is None:
+            raise KeyError("The 'is_active' field is not found in the user's info")
+        return target_status == UserStatus("live" if is_active else "deactivated")
```

Comment:
Avoid this type of functionality where the function takes an Enum and only accepts 2 possible values when this is framed as a generic test.

Can consider updating the /me endpoint (without breaking functionality) or have special handling for the other cases so it doesn't just always fail if someone wants to check UserStatus invited